src/server/components/dashboard/controllers.ts

```diff
@@ -349,3 +366,167 @@ export async function feedbackController(
+    });
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDateController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+function getMaxDate(list: List): Date {
+  const lastAnnualReview = list.jsonData.lastAnnualReviewDate ?? list.createdAt;
+  const lastAnnualReviewPlusYear = add(lastAnnualReview as Date, { years: 1 });
+  const maxDateFromLastAnnualReview = addSixMonths(lastAnnualReviewPlusYear);
+  const maxDateFromUserEnteredValues = addSixMonths(list.jsonData.annualReviewStartDate as number);
+  const maxDate = isBefore(maxDateFromUserEnteredValues, maxDateFromLastAnnualReview)
+    ? maxDateFromUserEnteredValues
+    : maxDateFromLastAnnualReview;
+
+  return maxDate;
+}
+
+function addSixMonths(date: number | string | Date): Date {
+  const annualReviewDate = (typeof date === "string") ? new Date(date) : date;
+  const newDate = add(annualReviewDate as Date, { months: 6 });
+
+  return newDate;
+}
+
+export async function listsEditAnnualReviewDatePostController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+
+    return (req.body.action === "confirmNewDate")
+      ? await confirmNewAnnualReviewDate(req, res)
+      : await updateNewAnnualReviewDate(req, res);
+
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDatePostController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+async function confirmNewAnnualReviewDate(req: Request, res: Response): Promise<void> {
+
+  const { listId } = req.params;
+  const list = await findListById(listId) as List;
+  const {day, month} = req.body;
+  const annualReviewDate = getAnnualReviewDate({day, month, list});
+
+  if (!annualReviewDate.isValid) {
+    req.flash('annualReviewError', annualReviewDate.errorMsg);
+    return res.redirect(`${dashboardRoutes.listsEditAnnualReviewDate.replace(":listId", list.id.toString())}`);
+  }
+
+  return res.render("dashboard/lists-edit-annual-review-date-confirm", {
+    ...DEFAULT_VIEW_PROPS,
+    newAnnualReviewDateFormatted: format(annualReviewDate.value as Date, DATE_FORMAT),
+    newAnnualReviewDate: annualReviewDate.value,
+    list,
+    csrfToken: getCSRFToken(req),
+  });
+}
+
+interface IsDateValidInput {
+  day: string,
+  month: string,
+  list: List
+}
+
+interface IsDateValidOutput {
+  isValid: boolean,
+  value: Date | null,
+  errorMsg: string
+}
+
+export function getAnnualReviewDate({day, month, list}: IsDateValidInput): IsDateValidOutput {
+  const lastAnnualReview = list.jsonData.lastAnnualReviewDate ?? list.createdAt;
+
+  const annualReviewYear = getAnnualReviewYear({day, month, lastAnnualReview: (list.jsonData.annualReviewStartDate ?? lastAnnualReview) as number});
+  const parsedDate = parse(`${month}/${day}/${annualReviewYear}`, "P", new Date());
+
+  const maxDate = getMaxDate(list);
+
+  const invalidResult = {isValid: false, value: null};
+  const isLeapYear = (): boolean => month === "2" && day === "29";
```

Comment:
this does not need to be a callable function

Reply:
Lol, I think we should come to a consensus on this https://github.com/UKForeignOffice/lists/pull/487#discussion_r1012919961

Reply:
Two similar but different issues here. But variable is correct for this instance 😄.

@alisalaman-cyb - your comment, I assume is more around intention and context rather than "it should be a function". 

The issue with readability was that `(month === "2" && day === "29")` is that you need to guess what the developer intended with this. 

To resolve that, you would just do `const isLeapYear = () => {..}`, or `const isLeapYear = month === "2" && day === "29";`.

let's call this guy anonymous function (because it is) 
`const isLeapYear = (): boolean => month === "2" && day === "29";`.
it's still a function.

so what if you refactored it to 
```.js
function isLeapYear() { // takes no arguments
 
}
```
does it still _feel_ right? even though it doesn't take any arguments, but gets the `month` and `day` variables from outside the fn? 

```.js
function myController() {

   const month = "1";
   const day = "2";
   
   function isLeapYear() { // takes no arguments
 
   }
}


``` 



`const isLeapYear = month === "2" && day === "29";` is better. 

further in the code, you just do `isLeapYear`. if it's `isLeapYear()` - you need to think about the context in which the (nested) function is called. and again.. just unnecessary. 

I would concede if the helper function is abstracted outside of the controller, and took parameters.

Reply:
well said.  Makes sense @jenbutongit